amethyst_assets/src/simple.rs

```diff
@@ -4,9 +4,80 @@
 use std::borrow::Cow;
 use std::error::Error;
 use std::marker::PhantomData;
+use std::sync::Arc;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+use parking_lot::RwLock;
 
 use {Asset, AssetSpec, Cache, Context};
 
+/// An `AssetPtr` which provides `push_update`, `update`
+/// and `is_shared` methods. These can simply be called
+/// in order to implement the `Asset` trait.
+///
+/// The recommended strategy is to create a struct for an
+/// asset which simply wraps `AssetPtr` and implements
+/// `Asset` by calling these methods. Methods for the asset
+/// can then be implemented on that wrapper struct by getting
+/// the inner asset with `inner` and `inner_mut`.
```

Comment:
Since A is recommended to be some kind of smart pointer can we include that in the docs?  Otherwise I could see people who don't know any better accidentally storing a copy of their asset multiple times in memory.  Something to the effect of "A will be cloned a lot, so keep it small.  Arc<T> would be a good fit here."